libs/vkd3d/resource.c

```diff
@@ -1122,6 +1130,12 @@ HRESULT vkd3d_get_image_allocation_info(struct d3d12_device *device,
      * align the image ourselves. */
     target_alignment = desc->Alignment ? desc->Alignment : D3D12_DEFAULT_RESOURCE_PLACEMENT_ALIGNMENT;
 
+    /* Tight alignment enforces small alignment for eligible resources */
+    if ((desc->Flags & D3D12_RESOURCE_FLAG_USE_TIGHT_ALIGNMENT) &&
+            d3d12_resource_supports_small_resource_alignment(desc, vkd3d_get_format(device, desc->Format,
+                    !!(desc->Flags & D3D12_RESOURCE_FLAG_ALLOW_DEPTH_STENCIL))))
+        target_alignment = D3D12_SMALL_RESOURCE_PLACEMENT_ALIGNMENT;
```

Comment:
Looking over this, I can't immediately see a code path for being able to expose the minimum image alignment to application. E.g. if driver only needs 256 byte alignment or something for an image, it seems like we at minimum return 4 KiB to application here?

Reply:
`target_alignment` is used to a) pad allocations if the driver-required alignment is *larger* and b) to select a tiling mode. If the driver gives us 256b alignment, this is irrelevant.